src/network/networkd-dhcp-common.c

```diff
@@ -64,15 +64,26 @@ void network_adjust_dhcp(Network *network) {
 }
 
 static struct DUID fallback_duid = { .type = DUID_TYPE_EN };
-DUID* link_get_duid(Link *link) {
-        if (link->network->duid.type != _DUID_TYPE_INVALID)
-                return &link->network->duid;
-        else if (link->hw_addr.length == 0 && IN_SET(link->manager->duid.type, DUID_TYPE_LLT, DUID_TYPE_LL))
+
+DUID* link_get_duid(Link *link, int family) {
```

Comment:
can we make the return struct `const`here btw? if so, we cann make fallback_duid const too, which sounds apropriate given it's actually constant, no?

Reply:
Done.